runtime/parachains/src/paras.rs

```diff
@@ -317,21 +317,24 @@ impl<T: Config> Module<T> {
 	pub(crate) fn initializer_finalize() { }
 
 	/// Called by the initializer to note that a new session has started.
-	pub(crate) fn initializer_on_new_session(_notification: &SessionChangeNotification<T::BlockNumber>) {
+	///
+	/// Returns the list of outgoing parachains for this session.
+	pub(crate) fn initializer_on_new_session(notification: &mut SessionChangeNotification<T::BlockNumber>) {
 		let now = <frame_system::Module<T>>::block_number();
-		let mut parachains = Self::clean_up_outgoing(now);
+		let (mut parachains, mut outgoing) = Self::clean_up_outgoing(now);
 		Self::apply_incoming(&mut parachains);
 		Self::apply_upgrades(&mut parachains);
 		Self::apply_downgrades(&mut parachains);
 		<Self as Store>::Parachains::set(parachains);
+		notification.outgoing_paras.append(&mut outgoing);
```

Comment:
I'd just assign it or `mem::replace` with assert. That's because we know that this field is assigned only by paras.

Reply:
I'm fine with making `initializer_on_new_session` return a `Vec<_>` for this module only, if that makes things easier.